src/Music/Tempo.cpp

```diff
@@ -0,0 +1,98 @@
+//
+// Created by adven on 2024/3/27.
+//
+#include "Music/Tempo.h"
+
+Tempo::Tempo() {}
+
+Tempo::~Tempo() {}
+
+void Tempo::readTempoFile(const std::string &txtFilePath) {
+    std::ifstream txtTempoFile(txtFilePath);
+    LOG_INFO("hello");
+
+    if (!txtTempoFile.is_open()) {
+        LOG_ERROR("fuck it can not open this file");
+        return;
+    }
+
+    std::string line;
+    std::getline(txtTempoFile, line);
+
+    m_tempoList = std::move(txtToVector(line, ','));
+
+    // 关闭文件流
+    txtTempoFile.close();
+}
+
+std::vector<std::size_t> Tempo::txtToVector(const std::string &line,
+                                            const char splitChar) {
+    std::stringstream ss(line);
+    std::string item;
+    std::size_t transform = 0;
+    std::vector<std::size_t> elems;
+    while (std::getline(ss, item, splitChar)) {
+        transform = std::stoi(item);
+        elems.push_back(std::move(transform));
+    }
+
+    m_tempoListLength = elems.size();
+    return elems;
+}
+
+bool Tempo::canBeClick(std::size_t offset) {
+
+    const auto tempoIndex = m_tempoIndex + m_punishTimes;
```

Comment:
m_tempoIndex [0, m_tempoListLength)
m_punishTimes [0,2]
const auto tempoIndex [0, m_tempoListLength+2)
It may cause index out of range
![image](https://github.com/NTUT-FUCK-PTSD/Fuck-PTSD/assets/25154413/66b0d611-202c-4031-ac5d-8368325b39f1)